src/scirpy/ir_dist/metrics.py

```diff
@@ -414,7 +362,217 @@ def _make_numba_matrix(distance_matrix: dict, alphabet: str = "ARNDCQEGHILKMFPST
+    def __init__(self, n_jobs: int = 1):
+        super().__init__()
+        self.n_jobs = n_jobs
+
+    @abc.abstractmethod
+    def _metric_mat(
+        self,
+        *,
+        seqs_mat1: np.ndarray,
+        seqs_mat2: np.ndarray,
+        seqs_L1: np.ndarray,
+        seqs_L2: np.ndarray,
+        is_symmetric: bool = False,
+        start_column: int = 0,
+    ) -> tuple[list[np.ndarray], list[np.ndarray], np.ndarray]:
+        pass
+
+    def _calc_dist_mat_block(
+        self,
+        seqs: Sequence[str],
+        seqs2: Optional[Sequence[str]] = None,
+        is_symmetric: bool = False,
+        start_column: int = 0,
+    ) -> csr_matrix:
+        """Computes a block of the final distance matrix and returns it as CSR matrix.
+        If the final result matrix that consists of all blocks together is symmetric, only the part
+        of the block that would contribute to the upper triangular matrix of the final result will be computed.
+        """
+        if len(seqs) == 0 or len(seqs2) == 0:
+            return csr_matrix((len(seqs), len(seqs2)))
+
+        max_seq_len = max(np.max([len(s) for s in seqs]), np.max([len(s) for s in seqs2]))
+        seqs_mat1, seqs_L1 = _seqs2mat(seqs, max_len=max_seq_len)
+        seqs_mat2, seqs_L2 = _seqs2mat(seqs2, max_len=max_seq_len)
+
+        data_rows, indices_rows, row_element_counts = self._metric_mat(
+            seqs_mat1=seqs_mat1,
+            seqs_mat2=seqs_mat2,
+            seqs_L1=seqs_L1,
+            seqs_L2=seqs_L2,
+            is_symmetric=is_symmetric,
+            start_column=start_column,
+        )
+
+        indptr = np.zeros(row_element_counts.shape[0] + 1)
+        indptr[1:] = np.cumsum(row_element_counts)
+        data, indices = np.concatenate(data_rows), np.concatenate(indices_rows)
+        sparse_distance_matrix = csr_matrix((data, indices, indptr), shape=(len(seqs), len(seqs2)))
+        return sparse_distance_matrix
+
+    def calc_dist_mat(self, seqs: Sequence[str], seqs2: Optional[Sequence[str]] = None) -> csr_matrix:
+        """Calculates the pairwise distances between two vectors of gene sequences based on the distance metric
+        of the derived class and returns a CSR distance matrix
+        """
+        if seqs2 is None:
+            seqs2 = seqs
+
+        seqs = np.array(seqs)
+        seqs2 = np.array(seqs2)
+        is_symmetric = np.array_equal(seqs, seqs2)
+        n_blocks = self.n_jobs * 2
+
+        if False:  # self.n_jobs > 1: --- only for intermediate version set to False
+            split_seqs = np.array_split(seqs, n_blocks)
+            start_columns = np.cumsum([0] + [len(seq) for seq in split_seqs[:-1]])
+            arguments = [(split_seqs[x], seqs2, is_symmetric, start_columns[x]) for x in range(n_blocks)]
+
+            delayed_jobs = [joblib.delayed(self._calc_dist_mat_block)(*args) for args in arguments]
+            results = list(_parallelize_with_joblib(delayed_jobs, total=len(arguments), n_jobs=self.n_jobs))
```

Comment:
You could directly use 
```
Parallel(return_as="list", n_jobs=self.n_jobs)(delayed_jobs)
```

here. The `_parallelize_with_joblib` wrapper is only there for the progressbar - and a progressbar doesn't make sense with this small number of jobs.